### ISSUE TITLE
Inconsistent behavior after setting moveSelection to true within sortable grid.

### ISSUE THREAD

**Issue by anasmi** [TatuLund/GridFastNavigation#76]
Steps to reproduce:
- Set moveSelection to true
- Sort the Integer(2) column
-Update any value in this column
-The focused and selected row are different. Navigating with keyboard inconsistent and restored only after reaching the header row and going back

Reproducible in the DemoFastGrid demo

**Comment by TatuLund** [comment]
Yes, this is a very interesting issue. Actually the root cause of the issue is not in the add-on, but it is somewhat due Grid's and its data communication behavior. It happens to be so, that if your Grid is sorted, the grid.getDataCommunictor.fetchItemsWithRange(rowIndex,1).get(0) will return item according to updated sorting, but you are not seeing the rows sorted according to edit until you have called grid.getDataProvider().refreshAll(). There is also a bug in add-on, which prevents it work correctly when Grid is sorted, even if you call grid.getDataProvider().refreshAll() e.g. in edit event. The getItem() method of the event will not return correct item, since it will return item before grid.getDataProvider().refreshAll() was called, not after. That should be fixable.
